packages/wallets/trezor/src/adapter.ts

```diff
@@ -0,0 +1,169 @@
+    supportedTransactionVersions: ReadonlySet<TransactionVersion> = new Set(['legacy', 0]);
+
+    private _derivationPath: string;
+    private _connectUrl?: string;
+    private _connecting: boolean;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ||
+        typeof document === 'undefined' ||
+        typeof navigator === 'undefined' ||
+        !navigator.hid
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.Loadable;
+
+    constructor(config: TrezorWalletAdapterConfig = {}) {
+        super();
+        this._derivationPath = config.derivationPath || `m/44'/501'/0'/0'`;
+        this._connectUrl = config.connectUrl;
+        this._connecting = false;
+        this._publicKey = null;
+    }
+
+    get publicKey() {
+        return this._publicKey;
+    }
+
+    get connecting() {
+        return this._connecting;
+    }
+
+    get readyState() {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            const connectUrl = this._connectUrl && this._connectUrl + (this._connectUrl.endsWith('/') ? '' : '/');
+
+            await TrezorConnect.init({
```

Comment:
This dependency will need to be `async import`ed. Check out how some of the other adapters that load external libraries do it.